crates/fuel-core/src/database.rs

```diff
@@ -299,36 +327,19 @@ impl Database {
         self.data.batch_write(&mut set.into_iter())
     }
 
-    fn remove<V: DeserializeOwned>(
+    fn take<V: DeserializeOwned>(
```

Comment:
did we rename this to take in case we'd want to have a separate remove fn that doesn't need to load the previous value?

Reply:
Yes, for example in `batch_write`. Plus in the case of clear sequential opcode. Plus later we can optimise VM execution to use it too in some cases.
<img width="842" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/5260f2fd-4cc9-4748-8409-1f351a0e0f55">